README.md

```diff
@@ -1,37 +1,264 @@
+users := mgmt.Management().V3().User()
 
+syncHandler := func(id string, obj *v3.User) (*v3.User, error) {
+	if obj != nil {
+		return obj, nil
+	}
+
+	recordedNote := obj.Annotations != nil && obj.Annotations["wroteanoteaboutuser"] == "true"
+
+	if recordedNote {
+		// there already is a note, noop
+		return obj, nil
+	}
+
+	// we are getting the "mainrecord" configmap from the configmap cache. The cache is maintained
+	// locally and can try to fulfill requests without using the k8s api. This is much faster and
+	// efficient, however it does not update immediately so it is possible that if the object
+	// being requested was recently created that the cache will miss and return a not found error.
+	// In this scenario you can either count on the handler reenqueueing and retrying or you can
+	// just use the regular client.
+	record, err := configmaps.Cache().Get("", "mainrecord")
+	if err != nil {
+		return obj, err
+	}
+
+	record.Data[obj.name] = "recorded"
+	record, err = configmaps.Update(record)
+	if err != nil {
+		return obj, err
+	}
+
+	// This is done because obj is from the cache that is iterated over to run handlers and peerform other tasks. If the subsequent
+	// update fails then we will end up with an object on our cache that does not match the "truth" (how the object is in etcd).
+	obj = obj.DeepCopy()
+
+	if obj.Annotations == nil {
+		obj.Anotations = make(map[string]string)
+	}
+
+	obj.Annotations["wroteanoteaboutuser"] = "true"
+
+	// Here we are using the k8s client embedded onto the users controller to perform an update. This will go to the K8s API.
+	return users.Update(obj)
+}
+
+users.OnChange(context.Background(), "user-example-annoate-note-handler", syncHandler)
+```
+### How to Run Handlers
+
+Now that we have registered an OnChange handler, we can run it like so:
+mgmt.Start(context.Background(), 50)
+
+<br>
+
+### Different Ways of Interacting With Objects
+In the above example, two clients and one cache are being used to interact with objects. A client can
+Create, Update, UpdatesStatus, Delete, Get, Watch and Patch an objects, or List and Watch object of its respective type. A Cache can get an object or list the objects for its respective type and will try to get the data locally (from its cache) if possible. The client and cache are the most common ways to interact with an object using wrangler.
+
+Another way to interact with objects is to use the Apply client. The apply client works similarly to applying yaml using kubectl. This has benefits such as not assuming the existence of an object like the Update method on a client does. Instead, you can apply a state and the object will be created if it does not exist already or be updated to match the passed desired state if the object does exist
+already. Apply also allows the use of multiple Owner References in a way unique from the client- if any owner reference is deleted the object will be deleted.
+
+<br>
+
+## A Look at Structures Used in Wrangler
 ```golang
```

Comment:
Could you add comments to all of these functions?

Reply:
Done.